analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+      ## tumor-immune
+      consensus_annotation %in% immune_labels &
+        stringr::str_starts(ewing_annotation, "tumor") ~ "tumor-immune",
+      ## tumor-endo
+      consensus_annotation %in% endo_labels &
+        stringr::str_starts(ewing_annotation, "tumor") ~ "tumor-endo",
+      ## tumor
+      stringr::str_starts(ewing_annotation, "tumor") ~ "tumor",
+      ## default
+      .default = ewing_annotation
+    )
+  ) |>
+  dplyr::select(barcodes, celltype)
+
+umap_df <- umap_df |>
+  dplyr::left_join(celltype_df)
+```
+
+
+## InferCNV Heatmap
+
+Below, we show the `inferCNV` heatmap for this library.
+To help contextualize this image, here is the table of known CNVs in Ewing sarcoma used for validation:
+
+```{r}
+validation_df
+```
+
+
+![InferCNV heatmap](`r infercnv_png`)
+
+
+## Distribution of CNVs per cell
+
+This section contains figures showing the total number of CNVs detected per cell.
+
+### CNVs in reference vs query cells
+
+The figures below compare the number of CNVs per cell between reference and query cells.
+Ideally, we'll see a bimodal distribution where reference cells are at/near zero, and there is a peak of higher values which may represent tumor cells.
+In the density plots below, distributions are colored based on whether the cells are in the reference or not (aka, "unknown" cells).
+
+```{r}
+# First, construct data frame of total `has_cnv` counts, with other variables needed for plots
+has_cnv_df <- infercnv_df |>
+  dplyr::select(
+    barcodes,
+    starts_with("has_cnv_")
+  ) |>
+  tidyr::pivot_longer(
+    starts_with("has_cnv_"),
+    names_to = "chr",
+    values_to = "cnv"
+  ) |>
+  # sum all values for each
+  dplyr::group_by(barcodes) |>
+  dplyr::summarize(total_cnv_per_cell = sum(cnv)) |>
+  dplyr::ungroup() |>
+  # bring back metadata
+  dplyr::inner_join(
+    infercnv_metadata_df,
+    by = "barcodes"
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df)
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+ggplot(has_cnv_df) +
+  aes(x = total_cnv_per_cell, fill = cell_group) +
+  geom_density(alpha = 0.5) +
+  theme(legend.position = "bottom")
+```
+
+### CNVs across cell types
+
+Next, we'll look at the per-cell distributions of CNVs across cell types, using labels derived from the `cell-type-ewings` module as follows:
+
+* `tumor-endo`: Cells labeled as tumor by cell-type-ewings, but endothelial by `cell-type-consensus`
+* `tumor-immune`: Cells labeled as tumor by cell-type-ewings, but immune by `cell-type-consensus`
+* `tumor`: Cells labeled as tumor by `cell-type-ewings`
+* `endo`: An endothelial cell type
+* `immune`: An immune cell type
+* If none of the above, their existing `cell-type-ewings` annotation
+
+Ideally, we'll see that tumor and Unknown cells from will have relatively higher CNV scores compared to other cell types, in particular those cell types in the reference.
+
+```{r fig.height=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
```

Comment:
I think it would be helpful to add the total number of cells for each annotation to the y-axis label. I also might sort by total number of cells.